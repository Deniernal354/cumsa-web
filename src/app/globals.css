@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
