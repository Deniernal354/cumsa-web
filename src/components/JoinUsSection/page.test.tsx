import { test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import JoinUs from "./page";

test("join us page snapshot", () => {
  const { asFragment } = render(<JoinUs />);
  expect(asFragment()).toMatchSnapshot();
});
