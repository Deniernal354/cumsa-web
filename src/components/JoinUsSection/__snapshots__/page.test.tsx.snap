// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`join us page snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-base py-12 flex flex-col gap-12 items-center"
  >
    <div
      class="w-full max-w-6xl rounded-xl shadow p-10 flex flex-col gap-8 items-center md:px-20 sm:px-6"
      style="background: rgb(230, 245, 248); box-shadow: 0 2px 24px 0 #2fbcd9ff;"
    >
      <h1
        class="text-3xl md:text-5xl font-extrabold text-center text-[var(--blue)] mb-2"
      >
        Want to be part of our MSA family?
      </h1>
      <p
        class="text-lg text-gray-700 text-center max-w-2xl"
      >
        Whether you want to 
        <span
          class="font-semibold text-[var(--red)]"
        >
          lead, help out, or just stay in the loop
        </span>
        , we would love to have you.
        <br />
        Join our council, subscribe to our newsletter, or join our Discord. Choose what feels right for you.
      </p>
      <div
        class="flex flex-col md:flex-row gap-6 w-full justify-center items-center"
      >
        <a
          class="px-7 py-3 rounded-full bg-[var(--blue)] text-white font-semibold text-lg shadow transition hover:opacity-90 min-w-[220px] text-center"
          href="https://docs.google.com/forms/d/e/1FAIpQLSelz_Ag6-8c5Fz5r0MqIC9urPbZrk5DLmHr_J8-ut0Cc3uJgw/viewform"
          rel="noopener noreferrer"
          target="_blank"
        >
          Become a Council Member
        </a>
        <a
          class="px-7 py-3 rounded-full bg-[var(--red)] text-white font-semibold text-lg shadow transition hover:opacity-90 min-w-[220px] text-center"
          href="https://discord.gg/GyV5tjQ5yw"
          rel="noopener noreferrer"
          target="_blank"
        >
          Join our Discord
        </a>
      </div>
      <div
        class="w-full flex flex-col items-center mt-6"
      >
        <h2
          class="text-2xl font-bold text-[var(--blue)] mb-2"
        >
          Stay in the Loop
        </h2>
        <p
          class="text-center text-gray-700 mb-4 max-w-lg"
        >
          Get the latest on events, prayer times, and more. No spam, just good vibes sent to your inbox every week.
        </p>
        <div
          class="w-full max-w-2xl"
        >
          <section
            class="py-8 md:py-12"
          >
            <div
              class="grid grid-cols-1 lg:grid-cols-5 gap-6 lg:gap-8 items-center"
            >
              <div
                class="lg:col-span-2 flex flex-col justify-center lg:pl-6"
              >
                <h2
                  class="text-2xl md:text-3xl lg:text-4xl font-bold leading-tight text-[var(--navy)]"
                >
                  Subscribe to the MSA Newsletter
                </h2>
              </div>
              <div
                class="lg:col-span-3 flex justify-center lg:justify-end"
              >
                <iframe
                  allowfullscreen=""
                  class="w-full max-w-md lg:max-w-lg"
                  height="520"
                  src="https://docs.google.com/forms/d/e/1FAIpQLSee5KLKC4gycv5aJBtdSpTr8hQVNT5oVY3Jpo_M2dG9-QCOjQ/viewform"
                  style="border: 0px; border-radius: 12px;"
                  title="Subscribe to the MSA Newsletter"
                />
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
